pkg/machines/components/vm/memoryModal.jsx

```diff
@@ -0,0 +1,230 @@
+import React from 'react';
+import {
+    Button,
+    HelpBlock,
+    Modal,
+} from 'patternfly-react';
+import PropTypes from 'prop-types';
+
+import cockpit from 'cockpit';
+import * as Select from 'cockpit-components-select.jsx';
+import { ModalError } from 'cockpit-components-inline-notification.jsx';
+import {
+    units,
+    convertToUnit,
+    toFixedPrecision,
+    vmId
+} from '../../helpers.js';
+import {
+    setMemory,
+    setMaxMemory,
+    getVm
+} from '../../actions/provider-actions.js';
+
+import './memoryModal.css';
+import 'form-layout.less';
+
+const _ = cockpit.gettext;
+
+export class MemoryModal extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            memory: props.vm.currentMemory, // Always in KiB
+            memoryUnit: units.MiB.name,
+            maxMemory: props.vm.memory, // Always in KiB
+            maxMemoryUnit: units.MiB.name,
```

Comment:
Are you sure that this is right? "KiB" vs. "MiB"? (The code down there converts to KiB) If it's right, it should at least get an explanation.

Reply:
Hi. Yes. memory and memoryUnit represent different things.

**memoryUnit** and **maxMemoryUnit** represent unit chosen by the user in UI. MiB is just initial value, user can choose between MiB and GiB.

**memory** and **maxMemory** represent values in our "backend", which are later passed to libvirt call, and it is more handy to keep them at same unit all the time. Otherwise we would have to convert them to same unit everytime we make same calculation or comparison between memory and maxMemory, as user can choose different units for different values. It's in KiB because that the unit in which are the values provided to us by libvirt and then again passed back to libvirt. So keeping it in KiB saves 4 possible unit conversions.

But yes, better comments are in place here.

Reply:
@skobyda actually this has changed, @martinpitt is correctly confused.
I found it very confusing how it was before, now memory is always what the user inputs which means it's following the units that are in memoryUnit. Same for maxMemory.
The comment it a leftover.
I will remove it.

Reply:
Right, that made me as well confused, with so many times being changed.

The comments are correct, we keep the state always in KB to make checks easier.
The initial memory unit as displayed to user is MiB, but we do the conversion from the KiB to the user selected memory Unit in the render.